ambari-server/src/main/java/org/apache/ambari/server/view/ViewRegistry.java

```diff
@@ -1135,6 +1135,12 @@ private boolean checkAutoInstanceConfig(AutoInstanceConfig autoConfig, StackId s
 
         String configStackId = autoConfig.getStackId();
 
+        if (configStackId.equals("*")) {
```

Comment:
Please move this inside the `if (configStackId != null)` block to prevent NPE if `configStackId` is missing.  (But keep it before the `new StackId(configStackId)` call, since `*` cannot  be parsed as `StackId`).

Reply:
Or just use StringUtils.equals() :)

Reply:
Hi @adoroszlai , @jonathan-hurley ,

Thanks for catching this.  I've  updated the patch to account for this, and have uploaded to the review.